cluster-autoscaler/core/scaledown/actuation/actuator_test.go

```diff
@@ -632,29 +749,34 @@ func TestStartDeletion(t *testing.T) {
 		"DS pods are evicted from empty nodes, but don't block deletion on error": {
 			emptyNodes: generateNodes(2, "empty"),
 			pods: map[string][]*apiv1.Pod{
-				"empty-node-0": {generateDsPod("empty-node-0-ds-pod-0", "empty-node-0"), generateDsPod("empty-node-0-ds-pod-1", "empty-node-0")},
-				"empty-node-1": {generateDsPod("empty-node-1-ds-pod-0", "empty-node-1"), generateDsPod("empty-node-1-ds-pod-1", "empty-node-1")},
+				"empty-node-0": {generateDsPod("ds-pod-0", "empty-node-0"), generateDsPod("ds-pod-1", "empty-node-0")},
+				"empty-node-1": {generateDsPod("ds-pod-0", "empty-node-1"), generateDsPod("ds-pod-1", "empty-node-1")},
 			},
 			failedPodDrain: map[string]bool{"empty-node-1-ds-pod-0": true},
 			wantStatus: &status.ScaleDownStatus{
 				Result: status.ScaleDownNodeDeleteStarted,
 				ScaledDownNodes: []*status.ScaleDownNode{
 					{
-						Node:        generateNode("empty-node-0"),
-						NodeGroup:   testNg,
+						Node:      generateNode("empty-node-0"),
+						NodeGroup: testNg,
+						// this is nil because DaemonSetEvictionForOccupiedNodes is
+						// not enabled for drained nodes in this test suite
 						EvictedPods: nil,
-						UtilInfo:    generateUtilInfo(2./8., 2./8.),
+						UtilInfo:    dsUtilInfo,
 					},
 					{
-						Node:        generateNode("empty-node-1"),
-						NodeGroup:   testNg,
+						Node:      generateNode("empty-node-1"),
+						NodeGroup: testNg,
+						// this is nil because DaemonSetEvictionForOccupiedNodes is
+						// not enabled for drained nodes in this test suite
 						EvictedPods: nil,
-						UtilInfo:    generateUtilInfo(2./8., 2./8.),
+						UtilInfo:    dsUtilInfo,
 					},
 				},
 			},
 			wantDeletedNodes: []string{"empty-node-0", "empty-node-1"},
-			wantDeletedPods:  []string{"empty-node-0-ds-pod-0", "empty-node-0-ds-pod-1", "empty-node-1-ds-pod-1"},
```

Comment:
This is not correct because no DS pods should be evicted since `DaemonSetEvictionForOccupiedNodes` is not enabled. Changed `wantDeletedPods` here to `nil`.

Reply:
This can be verified by checking
For drain nodes: [`prepareNodeForDeletion`](https://github.com/kubernetes/autoscaler/blob/4400d80ddb81fb570a5ef608c1bb4a8260b34d83/cluster-autoscaler/core/scaledown/actuation/actuator.go#L356) -> [`evictor.DrainNode`](https://github.com/kubernetes/autoscaler/blob/4373c467fe6f0a3b1a8beb9bc79e0ae1ee560445/cluster-autoscaler/core/scaledown/actuation/drain.go#L84) -> [`podsToEvict`-(https://github.com/kubernetes/autoscaler/blob/4373c467fe6f0a3b1a8beb9bc79e0ae1ee560445/cluster-autoscaler/core/scaledown/actuation/drain.go#L264) -> [`daemonset.PodsToEvict`](https://github.com/kubernetes/autoscaler/blob/0470fdfc354f72d51a4b5c382f552c0dff3fac4a/cluster-autoscaler/utils/daemonset/daemonset.go#L56)
For empty nodes: [`prepareNodeForDeletion`](https://github.com/kubernetes/autoscaler/blob/4400d80ddb81fb570a5ef608c1bb4a8260b34d83/cluster-autoscaler/core/scaledown/actuation/actuator.go#L360) -> [`evictor.EvictDaemonSetPods`](https://github.com/kubernetes/autoscaler/blob/4373c467fe6f0a3b1a8beb9bc79e0ae1ee560445/cluster-autoscaler/core/scaledown/actuation/drain.go#L185) -> [`daemonset.PodsToEvict`](https://github.com/kubernetes/autoscaler/blob/0470fdfc354f72d51a4b5c382f552c0dff3fac4a/cluster-autoscaler/utils/daemonset/daemonset.go#L56)